### PR TITLE
code cleanup part two: enforce checkstyle

### LINKED ISSUE #74: [new feature] checkstyle
this would be really nice to have to ensure we have consistent code

### PR DESCRIPTION
**Pull request by ix0rai**
- uses qsl's checkstyle with two changes:
   - removal of `JavadocStyle` module
   - updated import order to be everything and then `java` and `javax` packages
   - moved static imports to the bottom of the import list instead of the top (intellij default)
   - allows star imports on hamcrest's `Matchers` class and enigma's `TestEntryFactory`
- adds a checkstyle run to github actions
- most additions are simply throwing extra whitespace in
- fixes #74

### CODE REVIEW

**Comment by ix0rai** [comment]
failure is because I forgot to fix style in tests
whoops

File: config/checkstyle/checkstyle.xml [hunk 1]
```diff
@@ -0,0 +1,199 @@
...
+        <!--				- api packages: api + <not common> + client/server/nothing + <module-name> + 'v' + <version>-->
+        <!--				- impl+mixin packages : impl/mixin + <not common> + client/server/nothing + <module-name>-->
+        <!--				- literal legacy packages (exceptions)-->
+        <!--			- largely unconstained trailing subpackages-->
+        <!--			&ndash;&gt;-->
+        <!--			<property name="format"-->
+        <!--					  value="^net\.fabricmc\.fabric\.(api(?!\.common\.)(\.client|\.server|)(\.(?!client\.|server\.)[a-z]+([a-rt-z]|ss))+\.v[1-9][0-9]*|(impl|mixin|test)(?!\.common\.)(\.client|\.server|)(\.(?!client\.|server\.)[a-z]+[a-rt-z])+|api\.(event|util|biomes\.v1|registry|client\.screen|container|block|entity|client\.itemgroup|client\.keybinding|tag|tools|client\.model|network|server|client\.render|resource|client\.texture))(|\.[a-z]+(\.[a-z0-9]+)*)$"/>-->
+        <!--		</module>-->
+
+        <!--<module name="InvalidJavadocPosition"/>-->
```
Comment (sschr15) [inline config/checkstyle/checkstyle.xml:191]: Curious on the large comment block about Fabric package formatting

File: enigma-swing/src/main/java/cuchaz/enigma/gui/EditableType.java [hunk 2]
```diff
@@ -14,8 +14,7 @@ public enum EditableType {
 	FIELD,
 	PARAMETER,
 	LOCAL_VARIABLE,
-	JAVADOC,
-	;
+	JAVADOC;
```
Comment (sschr15) [inline enigma-swing/src/main/java/cuchaz/enigma/gui/EditableType.java:17]: Personal opinion: the previous style is nicer for future enum expansion

**Review by sschr15 — COMMENTED** [review]
wow that must've taken a lot of work

File: enigma-swing/src/main/java/cuchaz/enigma/gui/EditableType.java (hunk 2 above)
Comment (ix0rai) [reply enigma-swing/src/main/java/cuchaz/enigma/gui/EditableType.java:17]: I share your opinion, but I took this over disabling I think two inspections that errored here

File: config/checkstyle/checkstyle.xml (hunk 1 above)
Comment (ix0rai) [reply config/checkstyle/checkstyle.xml:191]: this checkstyle is qsl's, I think that comes straight from fapi

**Review by ByMartrixx — APPROVED** [review]
Looks good to me

**Pull request merged**